mantidproject/mantid
# Check version on startup

Added a new algorithms and wired in to check if a newer version is available on startup

release notes:

http://www.mantidproject.org/ReleaseNotes_3_5_UI_Changes#Setup_and_installation
Fixes #12064
### To test
1. check you are happy with the unit tests (these check the version comparison code, but the internet calls are stubbed)
2. Build MantidPlot yourself
   1. If you build it yourself then the call to check version is disabled in debug ,using the following line in the properties file (set it to 1 to work).  This will reset every time you build kernel.

```
CheckMantidVersion.OnStartup = 0
```
1. Start mantidplot
   1. set the logging in the gui to information
   2. close and start mantidplot again
   3. You should get some information out on startup about the current and versions available.
   4. In order to test things out in the gui you will need to manipulate the current Mantid version, and current release date.
   5. To do this you will need to edit MantidVersion.cpp and recompile
   6. you will need to change the values returned from MantidVersion::version() and MantidVersion::releaseDate()
      1, CheckMantidVersion will never consider releases from Github that are before the local releaseDate, and should only output versions above the local version number.
   7. Nightly build and dev releases ie. 3.4.20150703.1657, are considered to be 3.4.0 for the purposes of comparison.
